src/main/java/de/tum/in/ase/parser/strategy/CheckstyleFormatParser.java

```diff
@@ -0,0 +1,67 @@
+package de.tum.in.ase.parser.strategy;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import nu.xom.Document;
+import nu.xom.Element;
+
+import de.tum.in.ase.parser.domain.Issue;
+import de.tum.in.ase.parser.domain.Report;
+
+public abstract class CheckstyleFormatParser implements ParserStrategy {
+
+    protected static final String FILE_TAG = "file";
+    protected static final String FILE_ATT_NAME = "name";
+    protected static final String ERROR_ATT_SOURCE = "source";
+    protected static final String ERROR_ATT_SEVERITY = "severity";
+    protected static final String ERROR_ATT_MESSAGE = "message";
+    protected static final String ERROR_ATT_LINENUMBER = "line";
+    protected static final String ERROR_ATT_COLUMN = "column";
+
+    // The packages rooted at checks denote the category and rule
+    protected static final String CATEGORY_DELIMITER = "checks";
+    // Some rules don't belong to a category. We group them under this identifier.
+    protected static final String CATEGORY_MISCELLANEOUS = "miscellaneous";
```

Comment:
Can u move this to the checkstyle parser quickly?

Reply:
Sure. I'll do that